tests/loops/epoch/test_training_epoch_loop.py

```diff
@@ -141,3 +145,23 @@ def test_prepare_outputs_training_batch_end_manual(batch_end_outputs, expected):
         num_optimizers=-1,  # does not matter for manual optimization
     )
     assert prepared == expected
+
+
+def test_no_val_on_train_epoch_loop_restart(tmpdir):
+    """Test that training validation loop doesn't get triggered at the beginning of a restart."""
+    trainer_kwargs = {"max_epochs": 1, "limit_train_batches": 1, "limit_val_batches": 1, "enable_checkpointing": False}
```

Comment:
This should set 

```suggestion
    trainer_kwargs = {"max_epochs": 1, "limit_train_batches": 1, "limit_val_batches": 1, "num_sanity_val_steps": 0, "enable_checkpointing": False}
```

Reply:
we don't run a sanity check on a restart, so it would be fine?

Reply:
This doesn't matter for this PR but it will for #8578.

The reason is because here we are not "restarting" since the epoch was saved after the trainer completely finished (aka after `on_train_end`)

Reply:
we are restarting with more epochs... is it different?
quick ques: will restarting be set to True if I choose to train with more epochs after your PR?

Reply:
No. restarting only matters to know whether we are restarting from within an epoch or not.